<!DOCTYPE html>
<html lang="en">

<head>
	<meta charset="UTF-8">
	<title>TCuni</title>
  <!-- Bootstrap CSS File -->
  <link href="lib/bootstrap/css/bootstrap.min.css" rel="stylesheet">

  <!-- Libraries CSS Files -->
  <link href="lib/font-awesome/css/font-awesome.min.css" rel="stylesheet">
  <link href="lib/animate/animate.min.css" rel="stylesheet">
  <link href="lib/owlcarousel/assets/owl.carousel.min.css" rel="stylesheet">

  <!-- Main Stylesheet File -->
  <link href="css/style.css" rel="stylesheet">
</head>

<style type="text/css">


</style>

<body>

	<header id="header">
		<div class="container">

				<div id="logo" class="pull-left">
						<a href="#hero"><img src="imgs/logo.png" alt="" title="" /></img></a>
						<!-- Uncomment below if you prefer to use a text logo -->
						<!--<h1><a href="#hero">Regna</a></h1>-->
					  </div>	

			<nav class="nav-menu-container float-right">
				<ul class="nav-menu">
					<li class="menu-active"><a href="#hero">Home</a></li>
					<li class="menu-has-children">
						<a href="#">
							分析结果
						</a>
						<ul>
							<li><a href="#result-data">测序数据统计</a></li>
							<li><a href="#result-map">参考序列比对</a></li>
							<li><a href="#result-snp">SNP分析</a></li>
						</ul>
					</li>
					<li class="menu-has-children"><a href="#">项目流程</a>
						<ul>
							<li><a href="#intro-all">简介</a></li>
							<li><a href="#intro-lab">实验流程</a></li>
							<li><a href="#intro-analysis">数据分析流程</a></li>
						</ul>
					</li>
					<li class="menu-has-children">
						<a href="#">
							帮助及说明
						</a>
						<ul>
							<li><a href="#help-lib">建库测序</a></li>
							<li><a href="#help-dataqc">数据质控</a></li>
							<li><a href="#help-analysis">分析流程</a></li>
							<li><a href="#help-term">名词解释</a></li>
						</ul>
					</li>
				</ul>
			</nav>
		</div>
	</header>


	<!-- #intro -->

	<section id="hero">
			<div class="hero-container">
			  <h1>外显子组学</h1>
			  <h2>植物研究领域高效解决方案</h2>
			  <a href="#result-data" class="btn-get-started">Get Started</a>
			</div>
		  </section><!-- #hero -->

	<main>
		{% if seq_header %}
		<section id="result-data">
			<div class="container part">
				<!-- 结果内容 -->

				<!-- part3 -->
					<h3>原始数据质量控制</h3>
					<!-- part3-1 -->
					<div class="partP">
						<!-- <p class="haveBg" style="padding-left:30px;text-indent:0">本项目中，我们构建了6个DNA文库，使用Illumila Nova-seq平台进行测序，共获得了85.33G的总数据量，平均每个样品14.22G</p> -->

						<div>
							<table class="table table-striped table-hover">
								<thead class="thead-dark">
									<tr>
										{% for cell in seq_header[:-1] %}
										<th>{{cell}}</th>
										{% endfor %}
									</tr>
								</thead>
								<tbody>
									{% for row in seq_body %}
									<tr>
										{% for cell in row[:-1] %}
										<td>{{cell}}</td>
										{% endfor %}
									</tr>
									{% endfor %}
								</tbody>
							</table>
						</div>
						<!-- <div class="imgWrap">
						<a href="imgs/reads_quality.bar.report.png"><img src="imgs/reads_quality.bar.report.png" alt="reads_quality" class="maxWidth" style="max-width:600px"></a>
						<h6> 测序序列质量值分布图</h6>
					</div>
					<div class="imgWrap">
						<a href="imgs/gc_distribution.line.report.png"><img src="imgs/gc_distribution.line.report.png" alt="gc_distribution" class="maxWidth" style="max-width:600px"></a>
						<h6> GC 分布图</h6>
					</div> -->
					</div>


			</div>
		</section>
		{% endif %}

		{% if align_header %}
		<!-- part4 -->
		<section id="result-map">
			<div class="container part">
					<h3>参考序列比对分析</h3>

					<!-- part4-1 -->
					<div class="partP">
						<!-- <p class="haveBg" style="padding-left:30px;text-indent:0">本项目中，我们使用 <a href="http://rice.plantbiology.msu.edu/index.shtml"> Os-Nipponbare-Reference-IRGSP-1.0 Release 7</a> 为参考基因组，基因组大小为364M（参考序列中不含N）。</p> -->
						<div>
							<table class="table table-striped table-hover">
								<thead class="thead-dark">
									<tr>
										{% for cell in align_header %}
										<th>{{cell}}</th>
										{% endfor %}
									</tr>
								</thead>
								<tbody>
									{% for row in align_body %}
									<tr>
										{% for cell in row %}
										<td>{{cell}}</td>
										{% endfor %}
									</tr>
									{% endfor %}
								</tbody>
							</table>
						</div>
						<div class="imgWrap">


							<a href="imgs/Mapping_stats.png"><img src="imgs/Mapping_stats.png" alt="Mapping stats"
									class="maxWidth" style="max-width:600px"></a>
							<h6> Reads 与参考基因比对情况柱状图</h6>

						</div>

					</div>

			</div>
		</section>
		{% endif %}

		<section id="result-snp">
			<div class="container part">
				<!-- part5 -->
				<h3>SNP分析</h3>
				<!-- part5-1 -->

				{% if snp_summary_body %}
				<div class="partP">
					<h4 id="part5-1">统计Statistic</h4>
					<p class="haveBg">snp summary 结果如下:</p>
					<table class="table table-striped table-hover">
						<thead class="thead-dark">
							<tr>
								<th> Items </th>
								<th> Stat </th>
							</tr>
						</thead>
						<tbody>
							{% for row in snp_summary_body %}
							<tr>
								{% for cell in row %}
								<td>{{cell}}</td>
								{% endfor %}
							</tr>
							{% endfor %}
						</tbody>
					</table>
				</div>
				{% endif %}

				{% if snp_number_header %}
				<div class="partP">
					<p class="haveBg">SNP在染色体体上的统计</p>
					<table class="table table-striped table-hover">
						<thead class="thead-dark">
							<tr>
								{% for cell in snp_number_header %}
								<th>{{cell}}</th>
								{% endfor %}
							</tr>
						</thead>


						{% for row in snp_number_body %}
						<tr>
							{% for cell in row %}
							<td>{{cell}}</td>
							{% endfor %}
						</tr>
						{% endfor %}
					</table>
				</div>
				{% endif %}

				{% if snp_impact_body %}
				<div class="partP">
					<p class="haveBg">SNP变化的影响统计(Number of effects by impact)</p>
					<div class="imgWrap">
						{% if variant_summary %}
						<a href="imgs/Variant_stats_summary.png"><img src="imgs/Variant_stats_summary.png"
								alt="Variant rate by chromosome" class="maxWidth" style="max-width:900px"></a>
						{% endif %}
						<p class="haveBg"> Variate impact</p>
						<div>
							<table class="table table-striped table-hover">
								<thead class="thead-dark">
									<tr>
										<th> Type (alphabetical order) </th>
										<th> Count </th>
										<th> Percent </th>
									</tr>
								</thead>

								<tbody>
									{% for row in snp_impact_body %}
									<tr>
										{% for cell in row %}
										<td>{{cell}}</td>
										{% endfor %}
									</tr>
									{% endfor %}
								</tbody>
							</table>
							<br>
						</div>
					</div>
				</div>
				{% endif %}

				{% if snp_effect_body %}
				<div class="partP">
					<p class="haveBg">SNP在基因结构上的数据统计（Number of effects by type and region）</p>
					<table class="table table-striped table-hover">
						<thead class="thead-dark">
							<tr>
								<th> Type (alphabetical order) </th>
								<th> Count </th>
								<th> Percent </th>
							</tr>
						</thead>
						<tbody>
							{% for row in snp_effect_body %}
							<tr>
								{% for cell in row %}
								<td>{{cell}}</td>
								{% endfor %}
							</tr>
							{% endfor %}
						</tbody>
					</table>
				</div>
				{% endif %}

				{% if snp_region_body %}
				<div class="partP">
					<p class="haveBg">SNP位置(Number of effects by regin)</p>
					<table class="table table-striped table-hover">
						<thead class="thead-dark">
							<tr>
								<th> Type (alphabetical order) </th>
								<th> Count </th>
								<th> Percent </th>
							</tr>
						</thead>
						<tbody>
							{% for row in snp_region_body %}
							<tr>
								{% for cell in row %}
								<td>{{cell}}</td>
								{% endfor %}
							</tr>
							{% endfor %}
						</tbody>
					</table>
				</div>
				{% endif %}

				{% if varType_plots %}
				<div class="partP">
					<h4>Variants proportion by type</h4>
					<div id="testimonials" class="owl-carousel testimonials-carousel wow">
							{% for plot in varType_plots %}
							<div class="testimonial-item">
								<img src="imgs/{{plot}}" alt="">
							</div>
							{% endfor %}										
						</div>
				</div>
				{% endif %}

				{% if varRegion_plots %}
				<div class="partP">
					<h4>Number of effects by region</h4>
					<div id="testimonials" class="owl-carousel testimonials-carousel wow">
							{% for plot in varRegion_plots %}
							<div class="testimonial-item">
								<img src="imgs/{{plot}}" alt="">
							</div>
							{% endfor %}										
					</div>
				</div>
				{% endif %}

				{% if varImpact_plots %}
				<div class="partP">
					<h4>Proportion of effects by impact</h4>

					<div id="testimonials" class="owl-carousel testimonials-carousel wow">
						{% for plot in varImpact_plots %}
						<div class="testimonial-item">
							<img src="imgs/{{plot}}" alt="">
						</div>
						{% endfor %}										
					</div>
				</div>
				{% endif %}

				<div class="partP">
					<h4>上图中名词解释</h4>
					<table class="table table-striped table-hover">
						<thead class="thead-dark">
							<tr>
								<th> Type </th>
								<th> What is means </th>
								<th> Example </th>
							</tr>
						</thead>
						<tr>
							<td> SNP </td>
							<td> Single-Nucleotide Polymorphism </td>
							<td> Reference = 'A', Sample = 'C' </td>
						</tr>
						<tr>
							<td> Ins </td>
							<td> Insertion </td>
							<td> Reference = 'A', Sample = 'AGT' </td>
						</tr>
						<tr>
							<td> Del </td>
							<td> Deletion </td>
							<td> Reference = 'AC', Sample = 'C' </td>
						</tr>
						<tr>
							<td> MNP </td>
							<td> Multiple-nucleotide polymorphism </td>
							<td> Reference = 'ATA', Sample = 'GTC' </td>
						</tr>
						<tr>
							<td> MIXED </td>
							<td> Multiple-nucleotide and an InDel </td>
							<td> Reference = 'ATA', Sample = 'GTCAGT' </td>
						</tr>
					</table>

					<table class="table table-striped table-hover">
						<thead class="thead-dark">
							<tr>
								<th> Impact </th>
								<th> Meaning </th>
								<th> Example </th>
							</tr>
						</thead>
						<tr>
							<td> HIGH </td>
							<td> The variant is assumed to have high (disruptive) impact in the protein,
								probably causing protein truncation, loss of function or triggering
								nonsense
								mediated decay. </td>
							<td> stop_gained, frameshift_variant </td>
						</tr>
						<tr>
							<td> MODERATE </td>
							<td> A non-disruptive variant that might change protein effectiveness. </td>
							<td> missense_variant, inframe_deletion </td>
						</tr>
						<tr>
							<td> LOW </td>
							<td> Assumed to be mostly harmless or unlikely to change protein behavior.
							</td>
							<td> synonymous_variant </td>
						</tr>
						<tr>
							<td> MODIFIER </td>
							<td> Usually non-coding variants or variants affecting non-coding genes,
								where
								predictions are difficult or there is no evidence of impact. </td>
							<td> exon_variant, downstream_gene_variant </td>
						</tr>
					</table>

					<p class="haveBg">更多的名词解释，请查看<a
							href="#help-lib"><strong>帮助及说明</strong></a>。</p>
				</div>
			</div>
		</section>

		<section id="intro-all">
			<div class="container">
				<div class="part">
					<h3>简介</h3>
					<p class="haveBg" style="padding-left:30px;text-indent:0">
						获得样品后，如下图所示流程进行试验和数据处理，得到最终的分析结果<br>
						1) 对样品DNA进行随机打断建库.<br />
						2) 对捕获的文库使用Illumina测序平台进行测序.<br />
						3) 获得下机数据后使用GTAK最佳实践流程进行数据的标准call SNP分析得到vcf文件. <br>
						4) 使用个性化分析脚本进行数据的统计分析.<br />
					</p>
					<div class="imgWrap">
						<a href="imgs/p1.png"><img src="imgs/p1.png" alt="项目流程" class="maxWidth"
								style="max-height:220px"></a>
						<h6>图1：项目流程</h6>
					</div>
				</div>
			</div>
		</section>

		<section id="intro-lab">
			<div class="container">
				<div class="part">
					<h3>实验流程</h3>
					<p class="haveBg" style="padding-left:30px;text-indent:0">
						提取每个个体的基因组DNA并随机打断，电泳回收所需长度的DNA片段，并加上接头进行cluster制备，最后在Hiseq平台上机测序。</p>
					<div class="imgWrap">
						<a href="imgs/p2.png"><img src="imgs/p2.png" alt="生物信息分析流程" class="maxWidth"
								style="max-width:300px"></a>
						<h6>图2：建库流程</h6>
					</div>
				</div>
			</div>
		</section>

		<section id="intro-analysis">
			<div class="container">
				<!-- part2 -->
				<div class="part" id="part2">
					<h3>数据分析流程</h3>
					<p class="haveBg" style="padding-left:30px;text-indent:0">
						运用 BWA 软件将测序得到的 reads 与参考基因组序列进行比对。
						根据比对结果，计算出相对于参考基因组的测序深度和覆盖度。
						所测基因组的 SNPs、InDels 和结构变异 SVs 分别用 GATK、samtools、breakdancer 等软件进行检测。
						当有参考基因组的注释文件，可以分别在 SNP、InDel 和 SV 的数据集的基础上进行注释。</p>
					<p class="haveBg" style="padding-left:30px;text-indent:0">
						我们采用 <a href="https://www.broadinstitute.org" target="_blank">Broad Institute</a> 开发的变异检测流程
						<a href="https://www.broadinstitute.org/gatk/" target="_blank">GATK</a>，参考 GATK 的推荐流程 (<a
							href="https://software.broadinstitute.org/gatk/best-practices/bp_3step.php?case=GermShortWGS"
							target="_blank">Best Practices for Germline SNP & Indel Discovery in Whole Genome and Exome
							Sequence</a>)，
						对过滤后的测序序列进行参考基因组的比对分析.
						主要流程如下：<br /> (1) Map to reference (<a href="https://github.com/lh3/bwa" target="_blank">BWA</a>
						default parameters)</br>
						(2) Mark Duplicates (<a
							href="http://gatkforums.broadinstitute.org/gatk/discussion/2799#top">Picard</a>)</br>
						(3) Realign Indels (<a href="http://broadinstitute.github.io/picard/"
							target="_blank">Picard</a>)</br>
						(4) Recalibrate Base (<a href="http://broadinstitute.github.io/picard/"
							target="_blank">Picard</a>)</br>
						(5) Call Variants (<a href="https://www.broadinstitute.org/gatk/" target="_blank">GATK</a>)</br>
						(6) VCF Quality filter (<a href="https://www.broadinstitute.org/gatk/" target="_blank">GATK</a>
						standard)</br>
						(7) Annotating variants (<a href="http://snpeff.sourceforge.net/"
							target="_blank">snpEff</a>/Python Script)
					</p>
					<div class="imgWrap">
						<a href="imgs/BP_workflow_3.6.png"><img src="imgs/BP_workflow_3.6.png" alt="分析流程"
								class="maxWidth" style="max-height:520px"></a>
						<h6>图3：数据分析流程</h6>
					</div>
				</div>
			</div>
		</section>

		<section id="help-lib">
			<div class="container">
				<div class="part">
					<h3>建库测序</h3>

					<!-- part8-1 -->
					<div class="partP">
						<h4 id="part8-1">样品检测</h4>
						<p class="haveBg " style="padding-left:30px;text-indent:0 ">样品的检测主要包括4种方法：<br /> (1)
							琼脂糖凝胶电泳分析是否有污染<br /> (2) Qubit对浓度进行精确定量<br /> (3) Agilent 2100精确检测完整性<br /><br /> 主要仪器设备
							<br /> 1. PCR 扩增：Thermal cyclerS1000 (Bio-Rad)<br /> 2. 凝胶成像：Gel Doc XR+ (Bio-Rad)<br /> 3.
							qPCR 质控：StepOne Plus (ABI)<br /> 4. 测序：Illumina Nova-seq<br />
						</p>
					</div>
					<!-- part8-2 -->
					<div class="partP">
						<h4 id="part8-2">文库构建</h4>
						<p class="haveBg" style="padding-left:30px;text-indent:0 ">
							Total 4.5 mg of genomic DNA (35 ng/ml) was fragmented into 300 bp using the Bioruptor
							UCD-200 sonicator (Diagenode, Denville, NJ). Pre-libraries were constructed using the Kapa
							Hyper DNA library prep kit for Illumina HiSeq. Fragmented DNA (1 mg) was end repaired with
							an end-repair enzyme and a deoxyadenosine was added to the 30 end of the fragments. Kapa
							barcoded DNA and Kapa indexed adapters were ligated to the sample libraries. The
							adapter-ligated libraries were selected for an average insert size of 350 bp using
							next-generation sequencing cleanup and size select (NucleoMag, Macherey-Nagel, Duren,
							Germany) according to the manufacturer’s instructions. The pre-capture amplification was
							performed using the NumbleGen protocol with eight PCR cycles. Equal amounts of products from
							six libraries were pooled to obtain a total of 1 mg of DNA for the hybridization.
							Hybridization of sample libraries was performed at 47C for 60h~72 h using the SeqCap EZ
							library (Roche/NimbleGen, Madison, WI), probe custom designed with IWGSC RefSeq V1.0 genome.
							The hybrids were captured using the capture beads from the SeqCap EZ pure capture bead kit,
							washed, and amplified by ligationmediated PCR. The quality of the captured libraries was
							assessed using the Bioanalyzer 4200 instrument (Agilent Technologies, Santa Clara, CA). The
							libraries were quantified by quantitative real-time (qRT)-PCR and then sequenced using the
							Illumina HiSeq Nova platform to generate 150-bp paired-end reads.</p>
						<p class="haveBg " style="padding-left:30px;text-indent:0 ">The SeqCap EZ HyperCap workflow
							(Figure 1) involves:
							1. Preparation of the gDNA sample library using a KAPA Hyper Prep Kit or KAPA HyperPlus
							Library Preparation Kit.
							2. Amplification of the DNA sample library using KAPA HiFi HotStart ReadyMix (2X).
							3. Capturing target regions by hybridizing the gDNA sample library with the SeqCap EZ probe
							pool.
							4. Recovery of captured sample using the SeqCap Hybridization and Wash Kit.
							5. Amplification of the captured DNA sample using KAPA HiFi HotStart ReadyMix (2X).
							6. Sequencing the captured and amplified DNA sample using an Illumina sequencing instrument.
						</p>
						<div class="imgWrap">
							<a href="imgs/LibraryWorkflow.png"><img src="imgs/LibraryWorkflow.png " alt="文库构建"
									class="maxWidth" style="max-width:900px"></a>
							<h6>Exon Capture Library Workflow</h6>
						</div>
					</div>
					<!-- part8-3 -->
					<div class="partP">
						<h4 id="part8-3">双端测序</h4>
						<p class="haveBg " style="padding-left:30px;text-indent:0 ">
							Paired-end方法是指在构建待测DNA文库时在两端的接头上都加上测序引物结合位点，在第一轮测序完成后，去除第一轮测序的模板链，用对读测序模块(Paired-End
							Module)引导互补链在原位置再生和扩增，以达到第二轮测序所用的模板量，进行第二轮互补链的合成测序。</p>
					</div>
				</div>
			</div>
		</section>

		<section id="help-dataqc">
			<div class="container">
				<div class="part">
					<h3>原始数据质量控制说明</h3>
					<!-- part9-1 -->
					<div class="partP">
						<h4 id="part9-1">原始序列数据</h4>
						<p class="haveBg " style="padding-left:30px;text-indent:0">
							提供的文件格式一般为：项目编号/Flowcell号/样品名称_Lane号_I+index号.R1戒R2.clean.fastq.gz(例1_L5_I108.R1.clean.fastq.gz)
							提供的数据为压缩包形式，数据量比较大，解压缩后一般在window下不易打开，需要在linux下用less命令打开或者用其他编辑器打开。 数据格式的详细说明参考以下的网页：<a
								href="http://en.wikipedia.org/wiki/FASTQ_format ">http://en.wikipedia.org/wiki/FASTQ_format</a><br />
							FASTQ格式文件中每个read由四行描述组成，如下：
						</p>
						<div class="imgWrap ">
							<a href="imgs/p25.png "><img src="imgs/p25.png " alt="FASTQ格式文件图 " class="maxWidth "
									style="max-width:450px "></a>
							<h6> FASTQ格式文件图</h6>
						</div>
						<p class="haveBg " style="padding-left:30px;text-indent:0 ">
							第一行：以“@”开头，随后为Illumina 测序标识符(Sequence Identifiers)和index序列等；<br /> 第二行：即为测到的碱基序列；
							<br /> 第三行：以“+”开头，随后为Illumina 测序标识符(选择性部分)； <br /> 第四行：是对应碱基的测序质量，该行中每个字符对应的ASCII值减去33
							<br />
						</p>
						<p class="haveBg " style="padding-left:30px;text-indent:0 ">
							第四行为对应第二行碱基的测序质量值。如果测序错误率用e表示，illumina
							HiSeqTM2500/MiSeq的碱基质量值用Qphred表示，则有下列关系：<br />公式一：　Qphred　=　-10log10(e)</p>
					</div>
					<!-- part9-2 -->
					<div class="partP">
						<h4 id="part9-2">数据过滤</h4>
						<p class="haveBg " style="padding-left:30px;text-indent:0 ">
							通过设定一定的过滤标准，去除掉会对后续的信息分析造成干扰的接头信息、低质量碱基、未测出的碱基（以N表示）等。最终得到的数据，我们称之为clean data 或clean
							reads。我们的数据过滤标准为：<br />
							<span style="padding-left:30px;display: inline-block;text-indent: 0 ">
								1) 过滤掉含有接头序列（adaptor）的reads；<br />
								2) 当单端测序read中含有的N的含量超过该条read长度比例的3%时,去除此对pairedreads；<br />
								3) 当单端测序read中含有的低质量(低于3)碱基数超过该条read长度比例的50%时，去除此对paired reads。<br /></span>
						</p>
					</div>
					<!-- part9-3 -->
					<div class="partP">
						<h4 id="part9-3">测序错误率原因</h4>
						<p class="haveBg" style="padding-left:30px;text-indent:0 ">对于测序技术，测序错误率分布具有下面两个主要原因：<br />
							<span style="padding-left:30px;display: inline-block;text-indent: 0 ">
								1) 由于测序过程中化学试剂的消耗，导致测序错误率会随着测序序列(Sequenced Reads)长度的增加而升高。<br />
								2) PCR过程中随机引物和DNA模版的不完全结合(Jiang et al.)可能导致前几个碱基测序错误率较高。
							</span>
						</p>
					</div>
				</div>
			</div>
		</section>

		<section id="help-analysis">
			<div class="container">
				<div class="part">
					<h3>主要分析流程</h3>
					<!-- part10-1 -->
					<div class="partP ">
						<h4 id="part10-1">分析方法及软件版本</h4>
						<p class="haveBg " style="padding-left:30px;text-indent:0 ">
							Raw sequencing reads were processed using ‘Fastap’(v0.12.4) to remove low quality read and
							adapter.BWA (version 0.7.16) mem was applied to align the high quality reads to IWGSC RefSeq
							V1.0 genome with default parameters. samtools (version 1.9) were used to bam sort, remove
							read PCR duplications and sample alignment statistic. After that raw cohort vcf was worked
							out with GATK (version v4.0.10.1) module BaseRecalibrator, ApplyBQSR, HaplotypeCaller,
							CombineGVCFs and GenotypeGVCFs. The minimum-mapping-quality parameter was set as 30 for only
							high quality alignment reads used to call variants. bcftools (version 1.9) was performed as
							variants quality filtering with "QUAL>30" and "DP>5". To annotate variants, customized
							database contained IWGSC V1.1 HC/LC gene and own annotated gene were created by snpEff. </p>
					</div>
					<!-- part10-2 -->
					<div class="partP">
						<h4 id="part10-2">Reads 比对结果 IGV 可视化浏览</h4>
						<p class="haveBg" style="padding-left:30px;text-indent:0 ">
							我们提供 Reads在基因组上比对结果的 bam 格式文件，部分物种还提供相应的参考基因组和注释文件，
							并推荐使用 IGV (Integrative Genomics Viewer) 浏览器对 bam 文件进行可视化浏览。
							IGV 浏览器具有以下特点：(1)能在不同尺度下显示单个或多个读段在基因组上的位置，包括读段在各个染色体上的分布情况和在注释的外显子、内含子、剪接接合区、基因间区的分布情况等；
							(2)能在不同尺度下显示不同区域的读段丰度，以反映不同区域的转录水平；
							(3)能显示基因及其剪接异构体的注释信息；(4)能显示其他注释信息；
							(5)既可以从远程服务器端下载各种注释信息，又可以从本地加载注释信息。
							<a href="http://www.broadinstitute.org/software/igv/startingIGV " target="_blank ">参考 IGV
								浏览器使用方法</a>。
						</p>
						<div class="imgWrap">
							<a href="imgs/p28.png "><img src="imgs/p28.png " alt="IGV浏览器 " class="maxWidth "
									style="max-width:450px "></a>
							<h6> IGV 浏览器</h6>
						</div>
					</div>
				</div>
			</div>
		</section>

		<section id="help-term">
			<div class="container">
				<div class="part">
					<h3>名词解释</h3>

					<!-- part11-1 -->
					<div class="partP">
						<h4 id="part11-1">snpEff注释</h4>
						<table class="table table-striped table-hover">
							<thead class="thead-dark">
								<tr>
									<th> Effect<br>Seq. Ontology </th>
									<th> Effect<br>Classic </th>
									<th> Note &amp; Example </th>
									<th>Impact</th>
								</tr>
							</thead>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001580">
										coding_sequence_variant </a> </td>
								<td> CDS </td>
								<td> The variant hits a CDS. </td>
								<td> MODIFIER </td>
							</tr>
							<tr>
								<td> <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001580">chromosome</a>
								</td>
								<td> CHROMOSOME_LARGE DELETION </td>
								<td> A large part (over 1% or 1,000,000 bases) of the chromosome was deleted. </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:1000035">duplication</a>
								</td>
								<td> CHROMOSOME_LARGE_DUPLICATION </td>
								<td> Duplication of a large chromoome segment (over 1% or 1,000,000 bases). </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:1000036">inversion</a>
								</td>
								<td> CHROMOSOME_LARGE_INVERSION </td>
								<td> Inversion of a large chromoome segment (over 1% or 1,000,000 bases). </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001580">
										coding_sequence_variant </a> </td>
								<td> CODON_CHANGE </td>
								<td> One or many codons are changed <br>e.g.: An MNP of size multiple of 3 </td>
								<td> LOW </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001821">
										inframe_insertion </a> </td>
								<td> CODON_INSERTION </td>
								<td> One or many codons are inserted <br>e.g.: An insert multiple of three in a codon
									boundary </td>
								<td> MODERATE </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001824">
										disruptive_inframe_insertion</a> </td>
								<td> CODON_CHANGE_PLUS CODON_INSERTION </td>
								<td> One codon is changed and one or many codons are inserted <br>e.g.: An insert of
									size multiple of three, not at codon boundary </td>
								<td> MODERATE </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001822">
										inframe_deletion </a> </td>
								<td> CODON_DELETION </td>
								<td> One or many codons are deleted <br>e.g.: A deletion multiple of three at codon
									boundary </td>
								<td> MODERATE </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001826">
										disruptive_inframe_deletion </a> </td>
								<td> CODON_CHANGE_PLUS CODON_DELETION </td>
								<td> One codon is changed and one or more codons are deleted <br>e.g.: A deletion of
									size multiple of three, not at codon boundary </td>
								<td> MODERATE </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001632">
										downstream_gene_variant </a> </td>
								<td> DOWNSTREAM </td>
								<td> Downstream of a gene (default length: 5K bases) </td>
								<td> MODIFIER </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001791">
										exon_variant </a> </td>
								<td> EXON </td>
								<td> The variant hits an exon (from a non-coding transcript) or a retained intron. </td>
								<td> MODIFIER </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001572">
										exon_loss_variant </a> </td>
								<td> EXON_DELETED </td>
								<td> A deletion removes the whole exon. </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001572">exon_loss_variant</a>
								</td>
								<td> EXON_DELETED_PARTIAL </td>
								<td> Deletion affecting part of an exon. </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:1000035">duplication</a>
								</td>
								<td> EXON_DUPLICATION </td>
								<td> Duplication of an exon. </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:1000035">duplication</a>
								</td>
								<td> EXON_DUPLICATION_PARTIAL </td>
								<td> Duplication affecting part of an exon. </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:1000036">inversion</a>
								</td>
								<td> EXON_INVERSION </td>
								<td> Inversion of an exon. </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:1000036">inversion</a>
								</td>
								<td> EXON_INVERSION_PARTIAL </td>
								<td> Inversion affecting part of an exon. </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001589">
										frameshift_variant </a> </td>
								<td> FRAME_SHIFT </td>
								<td> Insertion or deletion causes a frame shift <br>e.g.: An indel size is not multple
									of 3 </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001564">
										gene_variant </a> </td>
								<td> GENE </td>
								<td> The variant hits a gene. </td>
								<td> MODIFIER </td>
							</tr>
							<tr>
								<td> <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001879">feature_ablation</a>
								</td>
								<td> GENE_DELETED </td>
								<td> Deletion of a gene. </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:1000035">duplication</a>
								</td>
								<td> GENE_DUPLICATION </td>
								<td> Duplication of a gene. </td>
								<td> MODERATE </td>
							</tr>
							<tr>
								<td> <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001565">gene_fusion</a>
								</td>
								<td> GENE_FUSION </td>
								<td> Fusion of two genes. </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001565">gene_fusion</a>
								</td>
								<td> GENE_FUSION_HALF </td>
								<td> Fusion of one gene and an intergenic region. </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:0002086">bidirectional_gene_fusion</a>
								</td>
								<td> GENE_FUSION_REVERESE </td>
								<td> Fusion of two genes in opposite directions. </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:0000904">rearranged_at_DNA_level</a>
								</td>
								<td> GENE_REARRANGEMENT </td>
								<td> Rearrengment affecting one or more genes. </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0000605">
										intergenic_region</a> </td>
								<td> INTERGENIC </td>
								<td> The variant is in an intergenic region </td>
								<td> MODIFIER </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0002017">
										conserved_intergenic_variant</a> </td>
								<td> INTERGENIC_CONSERVED </td>
								<td> The variant is in a highly conserved intergenic region </td>
								<td> MODIFIER </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0002011">
										intragenic_variant</a> </td>
								<td> INTRAGENIC </td>
								<td> The variant hits a gene, but no transcripts within the gene</td>
								<td> MODIFIER </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001627">
										intron_variant </a> </td>
								<td> INTRON </td>
								<td> Variant hits and intron. Technically, hits no exon in the transcript. </td>
								<td> MODIFIER </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0002018">
										conserved_intron_variant </a> </td>
								<td> INTRON_CONSERVED </td>
								<td> The variant is in a highly conserved intronic region </td>
								<td> MODIFIER </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0000276">
										miRNA </a> </td>
								<td> MICRO_RNA </td>
								<td> Variant affects an miRNA </td>
								<td> MODIFIER </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001583">
										missense_variant </a> </td>
								<td> NON_SYNONYMOUS_CODING </td>
								<td> Variant causes a codon that produces a different amino acid <br>e.g.: Tgg/Cgg, W/R
								</td>
								<td> MODERATE </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001582">
										initiator_codon_variant </a> </td>
								<td> NON_SYNONYMOUS_START </td>
								<td> Variant causes start codon to be mutated into another start codon (the new codon
									produces a different AA). <br>e.g.: Atg/Ctg, M/L (ATG and CTG can be START codons)
								</td>
								<td> LOW </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001567">
										stop_retained_variant </a> </td>
								<td> NON_SYNONYMOUS_STOP </td>
								<td> Variant causes stop codon to be mutated into another stop codon (the new codon
									produces a different AA). <br>e.g.: Atg/Ctg, M/L (ATG and CTG can be START codons)
								</td>
								<td> LOW </td>
							</tr>
							<tr>
								<td> <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001093">protein_protein_contact</a>
								</td>
								<td> PROTEIN_PROTEIN_INTERACTION_LOCUS </td>
								<td> Protein-Protein interacion loci. </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:0002093">structural_interaction_variant</a>
								</td>
								<td> PROTEIN_STRUCTURAL_INTERACTION_LOCUS </td>
								<td> Within protein interacion loci (e.g. two AA that are in contact within the same
									protein, prossibly helping structural conformation). </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0002008">
										rare_amino_acid_variant</a> </td>
								<td> RARE_AMINO_ACID </td>
								<td> The variant hits a rare amino acid thus is likely to produce protein loss of
									function </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001574">
										splice_acceptor_variant </a> </td>
								<td> SPLICE_SITE_ACCEPTOR </td>
								<td> The variant hits a splice acceptor site (defined as two bases before exon start,
									except for the first exon). </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001575">
										splice_donor_variant </a> </td>
								<td> SPLICE_SITE_DONOR </td>
								<td> The variant hits a Splice donor site (defined as two bases after coding exon end,
									except for the last exon). </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001630">
										splice_region_variant </a> </td>
								<td> SPLICE_SITE_REGION </td>
								<td> A sequence variant in which a change has occurred within the region of the splice
									site, either within 1-3 bases of the exon or 3-8 bases of the intron.</td>
								<td> LOW </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001630">
										splice_region_variant </a> </td>
								<td> SPLICE_SITE_BRANCH </td>
								<td> A varaint affective putative (Lariat) branch point, located in the intron.</td>
								<td> LOW </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001630">
										splice_region_variant </a> </td>
								<td> SPLICE_SITE_BRANCH_U12 </td>
								<td> A varaint affective putative (Lariat) branch point from U12 splicing machinery,
									located in the intron.</td>
								<td> MODERATE </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001578">
										stop_lost </a> </td>
								<td> STOP_LOST </td>
								<td> Variant causes stop codon to be mutated into a non-stop codon <br>e.g.: Tga/Cga,
									*/R </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001988">
										5_prime_UTR_premature start_codon_gain_variant </a> </td>
								<td> START_GAINED </td>
								<td> A variant in 5'UTR region produces a three base sequence that can be a START codon.
								</td>
								<td> LOW </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0002012">
										start_lost</a> </td>
								<td> START_LOST </td>
								<td> Variant causes start codon to be mutated into a non-start codon. <br> e.g.:
									aTg/aGg, M/R </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001587">
										stop_gained </a> </td>
								<td> STOP_GAINED </td>
								<td> Variant causes a STOP codon <br>e.g.: Cag/Tag, Q/* </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001819">
										synonymous_variant </a> </td>
								<td> SYNONYMOUS_CODING </td>
								<td> Variant causes a codon that produces the same amino acid <br>e.g.: Ttg/Ctg, L/L
								</td>
								<td> LOW </td>
							</tr>
							<tr>
								<td> <a href=" "> start_retained </a> </td>
								<td> SYNONYMOUS_START </td>
								<td> Variant causes start codon to be mutated into another start codon. <br>e.g.:
									Ttg/Ctg, L/L (TTG and CTG can be START codons) </td>
								<td> LOW </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001567">
										stop_retained_variant </a> </td>
								<td> SYNONYMOUS_STOP </td>
								<td> Variant causes stop codon to be mutated into another stop codon. <br>e.g.: taA/taG,
									*/* </td>
								<td> LOW </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001576">
										transcript_variant </a> </td>
								<td> TRANSCRIPT </td>
								<td> The variant hits a transcript. </td>
								<td> MODIFIER </td>
							</tr>
							<tr>
								<td> <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001879">feature_ablation</a>
								</td>
								<td> TRANSCRIPT_DELETED </td>
								<td> Deletion of a transcript. </td>
								<td> HIGH </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001566">
										regulatory_region_variant </a> </td>
								<td> REGULATION </td>
								<td> The variant hits a known regulatory feature (non-coding). </td>
								<td> MODIFIER </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001631">
										upstream_gene_variant </a> </td>
								<td> UPSTREAM </td>
								<td> Upstream of a gene (default length: 5K bases) </td>
								<td> MODIFIER </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001624">
										3_prime_UTR_variant </a> </td>
								<td> UTR_3_PRIME </td>
								<td> Variant hits 3'UTR region </td>
								<td> MODIFIER </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0002015">
										3_prime_UTR_truncation</a> + <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001572 ">
										exon_loss </a> </td>
								<td> UTR_3_DELETED </td>
								<td> The variant deletes an exon which is in the 3'UTR of the transcript </td>
								<td> MODERATE </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001623">
										5_prime_UTR_variant </a> </td>
								<td> UTR_5_PRIME </td>
								<td> Variant hits 5'UTR region </td>
								<td> MODIFIER </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0002013">
										5_prime_UTR_truncation</a> + <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001572 ">exon_loss_variant</a>
								</td>
								<td> UTR_5_DELETED </td>
								<td> The variant deletes an exon which is in the 5'UTR of the transcript </td>
								<td> MODERATE </td>
							</tr>
							<tr>
								<td> <a href="http://www.sequenceontology.org/browser/current_svn/term/SO:0002013">
										sequence_feature </a> + <a
										href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001572 ">exon_loss_variant</a>
								</td>
								<td> NEXT_PROT </td>
								<td> A 'NextProt' based annotation. Details are provided in the 'feature type' sub-field
									(ANN), or in the effect details (EFF). </td>
								<td> MODIFIER </td>
							</tr>
						</table>
					</div>
				</div>
			</div>
		</section>
	</main>

	<footer id="footer" class="section-bg">
		<div class="container">
			<div class="copyright">
				&copy; Copyright <strong>TCuni</strong>. All Rights Reserved
			</div>
		</div>
	</footer>
  <!-- JavaScript Libraries -->
  <script src="lib/jquery/jquery.min.js"></script>
  <script src="lib/jquery/jquery-migrate.min.js"></script>
  <script src="lib/bootstrap/js/bootstrap.bundle.min.js"></script>
  <script src="lib/easing/easing.min.js"></script>
  <script src="lib/wow/wow.min.js"></script>
  <script src="lib/waypoints/waypoints.min.js"></script>
  <script src="lib/counterup/counterup.min.js"></script>
  <script src="lib/owlcarousel/owl.carousel.min.js"></script>  
  <script src="lib/superfish/hoverIntent.js"></script>
  <script src="lib/superfish/superfish.min.js"></script>

  <!-- Contact Form JavaScript File -->
  <script src="contactform/contactform.js"></script>

  <!-- Template Main Javascript File -->
  <script src="js/main.js"></script>

</body>

</html>